jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleHelper.java

```diff
@@ -0,0 +1,699 @@
+				break;
+			case 2: /* getStatic */
+				result = lookup.findStaticGetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 3: /* putField */
+				result = lookup.findSetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 4: /* putStatic */
+				result = lookup.findStaticSetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 5: /* invokeVirtual */
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				result = lookup.findVirtual(referenceClazz, name, type);
+				break;
+			case 6: /* invokeStatic */
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				result = lookup.findStatic(referenceClazz, name, type);
+				break;
+			case 7: /* invokeSpecial */ 
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				result = lookup.findSpecial(referenceClazz, name, type, currentClass);
+				break;
+			case 8: /* newInvokeSpecial */
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				result = lookup.findConstructor(referenceClazz, type);
+				break;
+			case 9: /* invokeInterface */
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				result = lookup.findVirtual(referenceClazz, name, type);
+				break;
+			default:
+				/* Can never happen */
+				throw new UnsupportedOperationException();
+			}
+			return result;
+		} catch (IllegalAccessException iae) {
+			// Java spec expects an IllegalAccessError instead of IllegalAccessException thrown when an application attempts 
+			// (not reflectively) to access or modify a field, or to invoke a method that it doesn't have access to.
+			throw new IllegalAccessError(iae.getMessage()).initCause(iae);
+		}
+/*[ELSE]*/
+		try {
+			MethodHandles.Lookup lookup = new MethodHandles.Lookup(currentClass, false);
+			MethodType type = null;
+			MethodHandle result = null;
+
+			switch (cpRefKind) {
+			case 1: /* getField */
+				result = lookup.findGetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 2: /* getStatic */
+				result = lookup.findStaticGetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 3: /* putField */
+				result = lookup.findSetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 4: /* putStatic */
+				result = lookup.findStaticSetter(referenceClazz, name, resolveFieldHandleHelper(typeDescriptor, lookup, loader));
+				break;
+			case 5: /* invokeVirtual */
+				type = MethodType.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+				lookup.accessCheckArgRetTypes(type);
```

Comment:
If the only difference between the two cases - openj9 mh & openjdk mh - is this call, it's better to keep the 90% the same code the same and add preprocessor tags around this.

It also makes it clear that this is the piece that's different, and allows adding a comment to indicate the access checks haven't occurred here 

ie:
```
/*[IF OPENJDK_METHODHANDLES]*/
/* TODO: Investigate if an equivalent for lookup.accessCheckArgRetTypes is
 * needed for OJDK MH. If needed, find an alternative for
 * lookup.accessCheckArgRetTypes.
 */
/*[ELSE]*/
lookup.accessCheckArgRetTypes(type);
/*[ENDIF]*/
```

Reply:
Comment added. `sendResolveMethodHandle` and `resolveFieldHandleHelper` shared between J9 and OJDK MH using preprocessor flags.